pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/DynamicBrokerSelector.java

```diff
@@ -86,24 +87,20 @@ private void refresh() {
 
   @Nullable
   @Override
-  public String selectBroker(String table) {
-    if (table != null) {
-      String tableName =
-          table.replace(ExternalViewReader.OFFLINE_SUFFIX, "").replace(ExternalViewReader.REALTIME_SUFFIX, "");
-      List<String> list = _tableToBrokerListMapRef.get().get(tableName);
-      if (list != null && !list.isEmpty()) {
-        return list.get(RANDOM.nextInt(list.size()));
-      }
-      // In case tableName is formatted as <db>.<table>
-      int idx = tableName.indexOf('.');
-      if (idx > 0) {
-        tableName = tableName.substring(idx + 1);
+  public String selectBroker(List<String> tableNames) {
+    if (tableNames != null) {
+      List<List<String>> commonBrokers = new ArrayList<>();
```

Comment:
this logic seems similar to the one in brokercache do we plan to put them in utils too later?
also we don't really need to get the broker list for each table and then add to the commonBroker. we can simply call 1st table, then continuously call retainAll on the remaining table?

Reply:
The same happens `BrokerSelectorUtils.getTablesCommonBrokers` method.